### DISCUSSION TITLE
Not building for Mac Catalyst (GLKit not found)

### DISCUSSION THREAD

**Discussion by mkrn** [HaishinKit/HaishinKit.swift#1010] [Q&A] [answered]
**Describe the bug**
When building to Mac Catalyst, GLKit-related errors are raised. GLKit.h seems to be not available on Catalyst. For that reason GLHKView yields errors as well. 

**To Reproduce**
1. Add MacOS  Mac Catalyst in Deployment info of the project
2. Build

**Expected behavior**
Should build

**Screenshots**

**Desktop (please complete the following information):**
- Mac OS Monterrey 12.4 
- XCode 13.4.1

**Smartphone (please complete the following information):**

**Additional context**
It looks like we need to account for Mac Catalyst which is like IOS but need to add an additional check 
#if os(iOS) || os(tvOS)

**Comment by shogo4405** [comment]
It seemes old version of HaishinKit. https://github.com/shogo4405/HaishinKit.swift/commit/6d177aed6d0ddbb54225993619796bf032b38f0a

Is it 1.2.0 above?

**Comment by mkrn** [comment]
You are right.

Is HaishinKit supporting Mac Catalyst?

Also getting:
`Cannot find type 'UnsafeMutableAudioBufferListPointer' in scope` AudioCodec.AudioBuffer.swift

**Reply by floriangbh** [reply]
Hello, 

I have the same issue, with SPM : 

<img width="754" alt="Screenshot 2022-09-08 at 15 03 17" src="https://user-images.githubusercontent.com/14054997/189129048-6dd481e2-63e9-4acc-920e-e0954380974e.png">

Did you find a solution?

**Comment by shogo4405** [comment]
Yes. My computer works fine.

**Comment by mkrn** [comment, marked as answer]
/HaishinKit.swift/Carthage/Build/Logboard.xcframework:1:1: While building for Mac Catalyst, no library for this platform was found in 'HaishinKit.swift/Carthage/Build/Logboard.xcframework'.

Seems like carthage doesn't build Logboard for Catalyst any ideas?
Thanks

**Reply by shogo4405** [reply]
My example project works fine. 

> git clone https://github.com/shogo4405/HaishinKit.swift.git
> cd HaishinKit.swift
> carthage bootstrap --use-xcframeworks
> open HaishinKit.xcodeproj

Please use `--use-xcframeworks` options

<img width="1013" alt="スクリーンショット 2022-06-18 17 12 49" src="https://user-images.githubusercontent.com/810189/174429196-7e027468-0220-441a-a6c2-eabe05146796.png">

**Comment by mkrn** [comment]
What is your version of Carthage?
Mine is not building Catalyst xcframework 

Looks like it's not official support in carthage yet https://github.com/Carthage/Carthage/pull/3235 

I run:
carthage bootstrap --use-xcframeworks
Please update to the latest Carthage version: 0.39.0. You currently are on 0.38.0
*** Checking out Logboard at "2.3.0"
*** xcodebuild output can be found in /var/folders/56/c0ywsh9n6bbg4vycgglbkskh0000gn/T/carthage-xcodebuild.tIG5HJ.log
*** Building scheme "Logboard tvOS" in Logboard.xcworkspace
Could not find any available simulators for tvOS


![image](https://user-images.githubusercontent.com/1870075/233139762-3f7b6f99-b4f9-4e47-9d20-ca275d43c85f.png)

**Reply by shogo4405** [reply]
Certainly it doesn't work. I mistakenly thought "Mac (Designed for iPad)" was referring to Mac Catalyst. 

 <img width="377" alt="スクリーンショット 2023-04-20 22 19 12" src="https://user-images.githubusercontent.com/810189/233379178-7febd819-d824-494e-af80-deb3b5738ce1.png">

Logboard requires all simulators iOS, tvOS and watchOS.
`carthage bootstrap --use-xcframeworks`

**Comment by mkrn** [comment]
With carthage 0.39.0 getting this error in logboard:


/Users/mac/Sites/HaishinKit.swift/Carthage/Checkouts/Logboard/Logboard.xcodeproj: warning: The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.9, but the range of supported deployment target versions is 10.13 to 13.3.99. (in target 'Logboard macOS' from project 'Logboard')
** ARCHIVE FAILED **


The following build commands failed:
	Ld /Users/mac/Library/Caches/org.carthage.CarthageKit/DerivedData/14.3_14E222b/Logboard/2.3.0/Build/Intermediates.noindex/ArchiveIntermediates/Logboard\ macOS/IntermediateBuildFilesPath/Logboard.build/Release/Logboard\ macOS.build/Objects-normal/x86_64/Binary/Logboard normal x86_64 (in target 'Logboard macOS' from project 'Logboard')

**Reply by shogo4405** [reply]
Yes. Logboard 2.3.0 doesn't support Xcode 14.3. I fixed it Logboard 2.3.1.

**Comment by shogo4405** [comment]
Try a little harder, it seems that I can make it work. https://github.com/shogo4405/HaishinKit.swift/pull/1188
<img width="452" alt="スクリーンショット 2023-04-21 0 02 33" src="https://user-images.githubusercontent.com/810189/233407933-94f35a55-012e-43d9-9a48-3ba5b3c1a3a7.png">
